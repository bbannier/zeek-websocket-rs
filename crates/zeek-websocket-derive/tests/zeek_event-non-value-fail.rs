use zeek_websocket_derive::zeek_event;

#[zeek_event(handle_foo)]
fn foo(_xs: &u8) {}

fn main() {}
