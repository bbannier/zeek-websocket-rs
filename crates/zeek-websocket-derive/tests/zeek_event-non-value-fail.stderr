error[E0271]: type mismatch resolving `<&u8 as TryFrom<Value>>::Error == ConversionError`
 --> tests/zeek_event-non-value-fail.rs:3:1
  |
3 | #[zeek_event(handle_foo)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^ expected `ConversionError`, found `Infallible`
  |
  = note: this error originates in the attribute macro `zeek_event` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `&u8: From<Value>` is not satisfied
 --> tests/zeek_event-non-value-fail.rs:3:1
  |
3 | #[zeek_event(handle_foo)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<Value>` is not implemented for `&u8`
  |
  = help: the following other types implement trait `From<T>`:
            `u8` implements `From<Char>`
            `u8` implements `From<bool>`
            `u8` implements `From<tungstenite::protocol::frame::coding::OpCode>`
  = note: required for `Value` to implement `Into<&u8>`
  = note: required for `&u8` to implement `TryFrom<Value>`
  = note: required for `Value` to implement `TryInto<&u8>`
  = note: this error originates in the attribute macro `zeek_event` (in Nightly builds, run with -Z macro-backtrace for more info)
